include/swift/AST/DiagnosticsSema.def

```diff
@@ -5075,8 +5075,8 @@ ERROR(function_builder_infer_ambig, none,
 NOTE(function_builder_infer_add_return, none,
      "add an explicit 'return' statement to not use a function builder", ())
 NOTE(function_builder_infer_pick_specific, none,
-     "apply function builder %0 (inferred from protocol %1)",
-     (Type, DeclName))
+     "apply function builder %0 (inferred from %select{protocol|dynamic replacement of}1 %2)",
```

Comment:
When is it possible for there to be an ambiguity between a function builder type inferred from a protocol requirement and one inferred from dynamic replacement? Does this happen if the dynamic replacement satisfies a protocol requirement, and both the requirement and the dynamically replaced decl specify different function builders?

Reply:
Yeah, good point. I meant to have a test for that. I've added it in the latest commit.